accepted/9999-npm-audit-assertions.md

```diff
@@ -0,0 +1,64 @@
+  - `--impactful=<boolean>` this should just be `true` or `false`. If it's `true`, then an assertion is made that the advisory is impactful to the module passed to `--module`. If it's `false`, then an assertion is made that the advisory is not impactful to the module passed to `--module`.
+- `npm audit` will need to be updated to both consume additional data and provide additional filters.
+  - by default, `npm audit` should
+    - surface advisories that have no assertions or `--impactful=true` assertions
+    - ignore advisories that have `--impactful=false` assertions, **presuming** there are no alternative paths to the advisories that have no assertions or `--impactful=true` assertions
+  - the `--assertions` flag should be added
+    - it should take the following values:
+      - `only`: only show the advisories that have have assertions, `true` or `false`
+      - `only-true`: only show the advisories that have `--impactful=true` assertions
+      - `only-false`: only show the advisories that have `--impactful=false` assertions
+      - `unfiltered`: show all the advisories, regardless of whether they have assertions, `true` or `false`
+    - these should probably be sent to the server to be filtered server-side and enable smaller responses.
+  - the UI should also display if an assertion has been made, and what that assertion is.
+  - `--json` should include assertions, if any exist.
+- They adition of `npm audit assert` will require some kind of change to the server that serves the information that powers `npm audit`. Specifically, this information will need to be acceptted when `npm audit assert` is run, and returned with `npm audit` information. Additionally, depending on how this is implemented, this might reduce the number of requests made and data returned to end-users. Specifically, if the CLI assumes by default that users want to ignore vulnerabilities that have been marked as `--impactful=false` by transitive (or direct) dependencies **and** that is sent to the server, the server can return only what's needed rathter than the full set of data.
+
+## Rationale and Alternatives
+
+- Do nothing: The current state of the world that has caused [pain](https://overreacted.io/npm-audit-broken-by-design/).
+- Allow reporters to mark which modules are impacted: not really scalable, especially since reporters won't be experts in the same way that maintainers are.
+- Do something in a neutral space, run by an independent organization: There's no reason this couldn't eventually be implemented under this API. It is contingent on that work being done and being successfully adopted by others. That shouldn't necessarily prevent us from implementing this and mapping it over to that later.
+
```

Comment:
I think what this gets at is that there may other tools that will want to use the "asserted" data. The npm repository seems like a reasonable option for the data as it already the source of truth for the package itself. A few ideas off the top of my head:
* provide a way to get the asserted data for a module without using the npm client (existing UI?, REST enpoint?)
* Include the data when a module is installed/updated, that would like the local copy be used by other tools. 
allow other tools to leverage the data.

Reply:
I would assume this data would be included with existing package metadata and therefore accessible how people already access package metadata.

Reply:
Agree here mostly. To get the broader industry support I think this needs to be successful, we will need a well documented schema and set of api's.  This feedback is something I have heard from multiple people.

I think the schema and api to load these counter claims needs to be platform agnostic. I think that even "publish this to the npm registry" should be an implementation detail for npm. Preferably npm would just build a client for this contract, or at maybe we need an "extensions" to the api so that npm could say "instead of a GET request to your provider server, install package `foo` and read file `claims.json`".  Again, this is just rough ideas.